Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -344,6 +344,74 @@ class ChannelUpdater: Worker {
         }
     }
 
+    func createEphemeralMessage(
+        in cid: ChannelId,
+        messageId: MessageId?,
+        text: String,
+        pinning: MessagePinning? = nil,
+        isSilent: Bool,
+        command: String?,
+        arguments: String?,
+        attachments: [AnyAttachmentPayload] = [],
+        mentionedUserIds: [UserId],
+        quotedMessageId: MessageId?,
+        skipPush: Bool,
+        skipEnrichUrl: Bool,
+        poll: PollPayload? = nil,
+        extraData: [String: RawJSON],
+        completion: ((Result<ChatMessage, Error>) -> Void)? = nil
+    ) {
+        var newMessage: ChatMessage?
+        database.write({ (session) in
+            let newMessageDTO = try session.createNewMessage(
+                in: cid,
+                messageId: messageId,
+                text: text,
+                pinning: pinning,
+                command: command,
+                arguments: arguments,
+                parentMessageId: nil,
+                attachments: attachments,
+                mentionedUserIds: mentionedUserIds,
+                showReplyInChannel: false,
+                isSilent: isSilent,
+                quotedMessageId: quotedMessageId,
+                createdAt: nil,
+                skipPush: skipPush,
+                skipEnrichUrl: skipEnrichUrl,
+                poll: poll,
+                extraData: extraData
+            )
+            if quotedMessageId != nil {
+                newMessageDTO.showInsideThread = true
+            }
+            newMessageDTO.type = MessageType.regular.rawValue
+            newMessageDTO.localMessageState = .sending
```

Comment:
For testing proposes, I've used the `.sending` type, this ofc will have the Sending UI, but this is easily changed if we add a new state or a new property. This should also be discussed in the offical API.